github/main.js

```diff
@@ -1,552 +1,898 @@
-// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
-// License for the specific language governing permissions and limitations under
-// the License.
+'use strict';
 
 /**
- * @fileoverview Community Connector for GitHub. Retrieves historical star data
- * for a GitHub repository.
+ * @license
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0*
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
  */
 
-/** define namespace */
-var connector = connector || {};
-
-/** @const */
-connector.STARRED_AT = 'starred_at';
-
-/** @const */
-connector.LINK_KEY = 'Link';
-
-/** @const */
-connector.OAUTH_CONST = 'OAUTH2';
-
-/** @const */
-connector.OAUTH_CLIENT_ID = 'OAUTH_CLIENT_ID';
-
-/** @const */
-connector.OAUTH_CLIENT_SECRET = 'OAUTH_CLIENT_SECRET';
+/**
+ * HELPERS: THESE FUNCTIONS ARE GENERAL PURPOSE HELPERS FOR THE REST
+ * OF THE COMMUNITY CONNECTORS
+ */
 
-/** @const */
-connector.logEnabled = false;
+//TODO(nathanwest): Roll this stuff into a separate library
 
-/** @const */
-connector.defaultOrganization = 'google';
+/**
+ * Similar to Array.map, but for an object. Maps the value of each property to
+ * an array, in sorted order by property name.
+ * @param  {!Object<string, T>} object The object to map over
+ * @param  {!function(string, T, Object<string, T>): U} func The mapping
+ *   function. Called once with (property value, property name, object) for
+ *   each of the object properties, as returned by Object.keys
+ * @return {!Array<U>} The mapped array with the return values of func.
+ * @template T, U
+ * @example
+ * var obj = {c: 3, b: 2, a: 1}
+ * var result = mapObjectSorted(obj, function(value) { return value + 1})
+ * // result == [2,3,4]
+ */
+function mapObjectSorted(object, func) {
+  return Object.keys(object)
+    .sort()
+    .map(function(key) {
+      return func(object[key], key, object);
+    });
+}
 
-/** @const */
-connector.defaultRepository = 'datastudio';
+/**
+ * Turn a function into a singletpn function. A singleton function is
+ * only called once; subsequent calls always return the same object as
+ * the initial call. The underlying function is called with no arguments.
+ * The returned wrapper has a .reset() property attached, which may be
+ * called to reset the stored singleton value. This reset function returns
+ * that value.
+ *
+ * @param  {Function} func The function to wrap
+ * @return {Function}      The singleton wrapper
+ */
+function singleton(func) {
```

Comment:
I don't think this is actually improving performance since state doesn't stick around in between DS calls. It's fine to just call your `getOauthService` method every time it's needed.

Reply:
It wasn't actually performance I was worried about, but correctness; I couldn't find docs about how this works and wanted to be better safe than sorry. I can clear it out if you think it would be better.